arraycontext/container/arithmetic.py

```diff
@@ -343,6 +343,17 @@ def {fname}(arg1):
                                 else:
                                     raise ValueError(msg)""")
                     gen(f"return cls({zip_init_args})")
+                if _cls_has_array_context_attr:
+                    gen("if isinstance(arg2,"
```

Comment:
Ordering here matters a great deal. (Maybe there should be a comment stating this.) These cases should be sorted from most likely to least. Is this the second-most-likely case?

Reply:
IMO, this use-case would come just after host-scalars, as implemented in  93013c9.